### PR TITLE
Fix for issue # 10573

### PR DESCRIPTION
**Pull request by AbhiAgarwal192**

### CODE REVIEW

**Comment by ahsonkhan** [comment]
What issue is this fixing?

It doesn't look like # 10573 is related: https://github.com/dotnet/corefx/issues/10573

**Comment by AbhiAgarwal192** [comment]
> What issue is this fixing?
>  Please check for open issue #10573

> Can you also add a test please?
I have added the test case.

**Comment by ahsonkhan** [comment]
Ah, AspNetCore, gotcha :)

File: src/System.Text.Encodings.Web/tests/HtmlEncoderTests.cs [hunk 1]
```diff
@@ -13,18 +13,33 @@ namespace Microsoft.Framework.WebEncoders
...
-                System.Text.Encodings.Web.HtmlEncoder.Default.Encode(writer, "\U0001f4a9");
-                Assert.Equal("&#x1F4A9;", writer.GetStringBuilder().ToString());
+                System.Text.Encodings.Web.HtmlEncoder.Default.Encode(writer, actual);
+                Assert.Equal(expected, writer.GetStringBuilder().ToString());
             }
         }
-        
+
+        [Theory]
+        [InlineData("&#x1F602;2", "😂2")]
```
[the remark this answers is not in the document]
Comment (AbhiAgarwal192) [reply src/System.Text.Encodings.Web/tests/HtmlEncoderTests.cs]: Sure

**Comment by GrabYourPitchforks** [comment]
Hello Abhi, thank you for the PR!

Please be advised that this code base is about to undergo significant churn. The churn shouldn't affect this PR, but I wanted you to be aware of it in case you were considering submitting future PRs in this area, as I didn't want it to take you by surprise.

File: src/System.Text.Encodings.Web/tests/HtmlEncoderTests.cs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Licensed to the .NET Foundation under one or more agreements.
+﻿// Licensed to the .NET Foundation under one or more agreements.
```
[the remark this answers is not in the document]
Comment (AbhiAgarwal192) [reply src/System.Text.Encodings.Web/tests/HtmlEncoderTests.cs]: Sure

File: src/System.Text.Encodings.Web/tests/HtmlEncoderTests.cs [hunk 3]
```diff
@@ -13,18 +13,22 @@ namespace Microsoft.Framework.WebEncoders
 {
     public class HtmlEncoderTests
     {
-        [Fact]
-        public void TestSurrogate()
+        [Theory]
+        [InlineData("&#x1F4A9;", "\U0001f4a9")]
+        [InlineData("&#x1F602;2", "\U0001F6022")]
+        [InlineData("&#x1F602; 21", "\U0001F602 21")]
+        [InlineData("x&#x1F602;y", "x\U0001F602y")]
```
Comment (GrabYourPitchforks) [inline src/System.Text.Encodings.Web/tests/HtmlEncoderTests.cs]: One more test please, to capture another boundary condition. This tests an even-length input being provided to the `EncodeCore` method, not just to the `Encode` method.

`[InlineData("&#x1F602;x&#x1F602;y", "\U0001F602x\U0001F602y")]`

Comment (AbhiAgarwal192) [reply src/System.Text.Encodings.Web/tests/HtmlEncoderTests.cs]: Added and committed my changes.
